doc/doxygen/headers/concepts.h

```diff
@@ -0,0 +1,171 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2015 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+/**
+ * @defgroup Concepts Concepts, or expectations on template parameters
+ *
+ * Sometimes imposing constraints on the type of an object without requiring
+ * it to belong to a specific inheritance hierarchy is useful. These are
+ * usually referred to as <em>concepts</em> in the C++ community. This module
+ * lists the concepts commonly used in deal.II with brief descriptions of
+ * their intent. The convention in deal.II for listing constraints on a type
+ * is to provide the name of the concept as a <code>typename</code> in a
+ * template: for example, Vector depends on the underlying scalar field, so it
+ * is defined as
+ * @code
+ * template<typename Number>
+ * class Vector : public Subscriptor
+ * @endcode
+ * where Vector <em>is</em> a subscriptor, but depends on the parameter
+ * <code>Number</code> in a way that cannot be easily expressed with
+ * inheritance.
```

Comment:
I think this example is not a particularly good one since `Subscriptor` is a class few users will ever encounter.

The point you want to make is also not related to inheritance at all. The important issue is that `Vector` has a template argument `Number` that has to satisfy certain requirements ("concepts"), most notably that it is a scalar and that certain operations on it are allowed. I would just omit the derivation here altogether, and rewrite the sentence following the example code.